### PR TITLE
Fix terminal selection mouse-up with find overlay

### PR DESCRIPTION
**Pull request by lawrencecchen**
## Summary
- Keep terminal Ghostty mouse-button state owned by `GhosttyNSView` while selection is active.
- Forward drag and mouse-up events from the portal-hosted terminal find overlay back to the terminal when the terminal has a pending left-button release.
- Add regression coverage for the find-overlay-captured mouse-up path.

## Testing
- `xcodebuild -project cmux.xcodeproj -scheme cmux -configuration Debug -destination 'platform=macOS' -derivedDataPath /tmp/cmux-task-selection-mouseup-find-overlay build` passed
- AWS targeted unit test attempted with `cmux-unit` and `-only-testing:cmuxTests/GhosttySurfaceOverlayTests/testSearchOverlayForwardsTerminalMouseReleaseDuringSelectionDrag`, but the runner failed during package checkout with `No space left on device` before running tests.

## Issues
- Plain-text task: fix terminal selection mouse-up loss when Cmd+F find overlay is open

---
<a href="https://app.blacksmith.sh/manaflow-ai/codesmith/cmux/pr/5335"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-dark-v2.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-light-v2.svg">View with Codesmith</picture></a> <a href="https://backend.blacksmith.sh/track/enable-autofix?expires=1783135766&installation_id=133855650&pr_number=5335&repository=manaflow-ai%2Fcmux&return_to=https%3A%2F%2Fgithub.com%2Fmanaflow-ai%2Fcmux%2Fpull%2F5335&signature=5503828e1c055b2ea0ad9af47947b1975acae7faa0003844f15956e69a76d29f"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-light.svg">Autofix with Codesmith</picture></a>
<sup>Need help on this PR? Tag <code>/codesmith</code> with what you need. Autofix is disabled.</sup>

---

> [!NOTE]
> **Low Risk**
> Localized AppKit mouse routing for find overlay and terminal selection; covered by new unit tests with no auth or data-path changes.
> 
> **Overview**
> Fixes **stuck terminal text selection** when the user releases the mouse over the **Cmd+F find overlay** instead of the terminal surface.
> 
> `GhosttyNSView` now tracks **`hasPendingLeftMouseRelease`** after left `mouseDown` and routes release handling through **`completePendingLeftMouseRelease`**, which still clears the flag if the Ghostty surface is already gone. **`forwardPendingLeftMouseDrag`** keeps pointer updates flowing during drags that cross the overlay.
> 
> The find UI is hosted in a new **`TerminalSearchOverlayHostingView`** that forwards **`mouseDragged`** / **`mouseUp`** to the terminal when a pending release exists. Regression tests cover overlay-captured release and cleanup after surface teardown.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 4a18ba507a84feca3c2c90d8c7323b180c2feb83. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

---
## Summary by cubic
Fixes lost terminal selection mouse-up when the Cmd+F find overlay is open. Selections finish even if the mouse is released over the overlay, and the pending-release state now clears reliably.

- **Bug Fixes**
  - Track left-button ownership in `GhosttyNSView` with a pending-release flag; forward drag via `forwardPendingLeftMouseDrag` and centralize release in `completePendingLeftMouseRelease`.
  - Use `TerminalSearchOverlayHostingView` to forward drag/up to the terminal when a release is pending.
  - Add Swift Testing regressions for overlay-captured release and cleanup when the `Ghostty` surface is gone.

<sup>Written for commit 4a18ba507a84feca3c2c90d8c7323b180c2feb83. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/manaflow-ai/cmux/pull/5335?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed mouse event handling in the search overlay to properly defer and resolve command-click operations in the correct sequence.
  * Improved state management for mouse button release events during search overlay interactions.

* **Tests**
  * Added test suite for mouse release event forwarding in the search overlay, validating correct behavior during active searches and edge cases involving surface cleanup.

### CODE REVIEW

File: Sources/GhosttyTerminalView.swift [hunk 1]
```diff
@@ -10030,16 +10031,33 @@ class GhosttyNSView: NSView, NSUserInterfaceValidations {
...
+        let eventPoint = convert(event.locationInWindow, from: nil)
+        trackMousePointIfUsable(eventPoint)
+        ghostty_surface_mouse_pos(surface, eventPoint.x, bounds.height - eventPoint.y, modsFromEvent(event))
+        return true
+    }
+
+    @discardableResult
+    fileprivate func completePendingLeftMouseRelease(with event: NSEvent) -> Bool {
+        guard hasPendingLeftMouseRelease, let surface else { return false }
+        hasPendingLeftMouseRelease = false
```
[the remark this answers is not in the document]
Comment (lawrencecchen) [reply Sources/GhosttyTerminalView.swift:10056]: Fixed by clearing the pending left-mouse release flag before checking whether the Ghostty surface is still live. I also added coverage for the surface-release edge case.

*— Claude Code*

File: Sources/GhosttyTerminalView.swift [hunk 2]
```diff
@@ -10030,16 +10031,33 @@ class GhosttyNSView: NSView, NSUserInterfaceValidations {
...
         let point = convert(event.locationInWindow, from: nil)
         let consumed = ghostty_surface_mouse_button(surface, GHOSTTY_MOUSE_RELEASE, GHOSTTY_MOUSE_LEFT, modsFromEvent(event))
         _ = handleCommandClickRelease(at: point, modifierFlags: event.modifierFlags, ghosttyConsumed: consumed)
+        return true
     }
```
[the remark this answers is not in the document]
Comment (lawrencecchen) [reply Sources/GhosttyTerminalView.swift]: Fixed by clearing the pending left-mouse release flag before checking whether the Ghostty surface is still live. I also added coverage for the surface-release edge case.

*— Claude Code*

File: Sources/GhosttyTerminalView.swift [hunk 3]
```diff
@@ -10030,16 +10031,33 @@ class GhosttyNSView: NSView, NSUserInterfaceValidations {
...
         #if DEBUG
         cmuxDebugLog("terminal.mouseUp surface=\(terminalSurface?.id.uuidString.prefix(5) ?? "nil") mods=[\(debugModifierString(event.modifierFlags))]")
         #endif
-        guard let surface = surface else { return }
+        completePendingLeftMouseRelease(with: event)
+    }
+
+    @discardableResult
+    fileprivate func forwardPendingLeftMouseDrag(with event: NSEvent) -> Bool {
+        guard hasPendingLeftMouseRelease, let surface else { return false }
```
[the remark this answers is not in the document]
Comment (lawrencecchen) [reply Sources/GhosttyTerminalView.swift:10046]: Fixed by clearing the pending left-mouse release flag before checking whether the Ghostty surface is still live. I also added coverage for the surface-release edge case.

*— Claude Code*
